### ISSUE TITLE
[ISSUE]: Strange bug with put command and txt file

### ISSUE THREAD

**Issue by GDX-X** [uyjulian/pfsshell#41]
### Checks

- [X] I have checked existing issues for duplicates and found none

### Describe the issue

I found a strange bug on pfsshell (present on all versions)

the steps to take to reproduce the bug

1) Create a txt file

2) write `hdd_partition=__common`

3) put it on hard drive

4) Edit the txt file and replace `__common` with `+OPL`

5) put it on hard drive

6) recover file from hard drive

who will give
`hdd_partition=+OPLmmon`

I also tried deleting the file before copying it, it does the same thing.

https://www.youtube.com/watch?v=CytM9ilctwQ

### Console model

X

**Comment by uyjulian** [comment]
Note to self: ftruncate?

**Comment by uyjulian** [comment]
Due to the fact that `truncate` is not implemented, this will require a large rework of the code to support properly.  

For now, save the changes to a new file instead of doing in-place changes.

**Comment by GDX-X** [comment]
Yeah, but it's not too disturbing. because the file will be replaced in any case as long as it works it is not urgent. the solution is simply to quit pfshell and restart it to have the complete file
